### PR TITLE
Remove freezed&either

### PR DESCRIPTION
**Pull request by yassmeensamy**

### CODE REVIEW

File: .vscode/launch.json
Comment (Abdelazeem777) [inline .vscode/launch.json:1]: Please don't add such personal configration files like this one on the remote repo

Better add it to gitIgnore

File: lib/app/router.dart [hunk 1]
```diff
@@ -23,8 +22,6 @@ Route<dynamic>? onGenerateRoute(RouteSettings settings) {
 
   if (name == InitView.route) {
     page = InitView();
-  } else if (name == OnboardingView.route) {
-    page = const OnboardingView();
```
Comment (Abdelazeem777) [inline lib/app/router.dart:27]: So we don't have this page anymore?

File: lib/features/authentication/data/models/verify_otp_response_model.dart [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-  }) = _VerifyOtpResponseModel;
+  VerifyOtpResponseModel({
+    required this.accessToken,
+    required this.refreshToken,
+    required this.customer,
+  });
 
-  factory VerifyOtpResponseModel.fromJson(Map<String, dynamic> json) =>
-      _$VerifyOtpResponseModelFromJson(json);
+  factory VerifyOtpResponseModel.fromJson(Map<String, dynamic> json) {
```
Comment (Abdelazeem777) [inline lib/features/authentication/data/models/verify_otp_response_model.dart:14]: Hey hey, We agreed about the difference between the `Json` and `Map`

So this method must be `fromMap`

File: lib/app/ServiceLocator.dart
Comment (Abdelazeem777) [inline lib/app/ServiceLocator.dart:1]: file names must be in snake_case!

File: lib/features/authentication/presentation/views/screens/login/login_view_model.dart [hunk 3]
```diff
@@ -30,69 +33,58 @@ class LoginViewModel extends Cubit<LoginViewState>
...
-      ),
-
-    );
-   }
+      );
+    } on ServerException catch (e) 
+    {
+      
+      //emit(state)
+    }
```
Comment (Abdelazeem777) [inline lib/features/authentication/presentation/views/screens/login/login_view_model.dart:48]: where is the handling here?

We need to emit an error state here!

File: lib/features/authentication/presentation/views/screens/login/login_view_model.dart [hunk 4]
```diff
@@ -30,69 +33,58 @@ class LoginViewModel extends Cubit<LoginViewState>
...
+
+    try {
+      final selection = await SelectCountryModal(
+        countries: countries,
+        selectedCountry: state.selectedCountry!,
+      ).show(context);
+      if (selection != null) {
+        emit(state.copyWith(selectedCountry: selection));
+      }
+    } catch (e) {}
```
Comment (Abdelazeem777) [inline lib/features/authentication/presentation/views/screens/login/login_view_model.dart:66]: Same we need to handle the error states!

File: lib/features/google_map/data/data_sources/googlemap_ApiRemoteDataSource.dart [hunk 5]
```diff
@@ -1,39 +1,35 @@
...
-class GoogleMapApiRemoteDataSource implements GoogleMapRemoteDataSource {
+class GoogleMapApiRemoteDataSource implements GoogleMapRemoteDataSource 
+{
   final DioApiClient _dioClient;
-  GoogleMapApiRemoteDataSource({required DioApiClient dioClient})
-      : _dioClient = dioClient;
+    final Dio _dio;
+  GoogleMapApiRemoteDataSource({required DioApiClient dioClient ,required Dio dio})
+      : _dioClient = dioClient,
+      _dio = dio;
```
Comment (Abdelazeem777) [inline lib/features/google_map/data/data_sources/googlemap_ApiRemoteDataSource.dart:13]: Why do we have 2 dios?

File: lib/features/home/data/data_sources/HomeData-api_remote_data_source.dart [hunk 6]
```diff
@@ -1,38 +1,29 @@
...
-import 'package:eram_express/features/home/data/models/cargo-subcategoryModel.dart';
-import 'package:eram_express/features/home/data/models/goods-typeModel.dart';
-import 'package:eram_express_shared/core/api/api_endpoint.dart';
-import 'package:eram_express_shared/core/api/api_error.dart';
+
+import 'dart:io';
+import 'package:eram_express/features/home/data/models/home-Model.dart';
 import 'package:eram_express_shared/core/api/dio_api_client.dart';
 
+import 'homeData-api-endpoint.dart';
```
Comment (Abdelazeem777) [inline lib/features/home/data/data_sources/HomeData-api_remote_data_source.dart:6]: Wrong filename!

File: lib/features/home/data/data_sources/HomeData-api_remote_data_source.dart
Comment (Abdelazeem777) [inline lib/features/home/data/data_sources/HomeData-api_remote_data_source.dart:1]: rename this file

we are using snake_case in the file naming!!

File: lib/features/init/presentation/views/init_view.dart [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
 
 class InitView extends StatelessWidget {
   static const String route = '/';
 
   final viewModel = InitViewModel(
-    authenticationRepository: authenticationRepository,
-    configurationsRepository: configurationsRepository,
+    authenticationRepository: sl<AuthenticationRepository>(),
+    configurationsRepository: sl<ConfigurationsRepository>(),
   );
```
Comment (Abdelazeem777) [inline lib/features/init/presentation/views/init_view.dart:17]: Hey, why are we still using the normal ViewModel creation here?

File: lib/features/google_map/data/data_sources/googlemap_ApiRemoteDataSource.dart (hunk 5 above)
Comment (yassmeensamy) [reply lib/features/google_map/data/data_sources/googlemap_ApiRemoteDataSource.dart]: dioClient  now become NetworkService and contains the mainUrl, but I use dio in the Google Maps requests

File: lib/app/router.dart (hunk 1 above)
Comment (yassmeensamy) [reply lib/app/router.dart:27]: We'll need it when merging the onboarding branch, but this page isn't in this branch.

**Pull request closed without merging**